public/app/features/plugins/sql/components/QueryEditor.tsx

```diff
@@ -78,8 +105,36 @@ export function SqlQueryEditor({ datasource, query, onChange, onRunQuery, range,
 
   return (
     <>
+      {hasDatabaseConfigIssue && (
+        <Alert
+          severity="warning"
+          title="Default datasource update"
+          elevated={true}
+          onRemove={() => {
+            // Remove the warning, and reset state with the new database.
+            setHasDatabaseConfigIssue(false);
+            onChange({ ...query, dataset: preconfiguredDatabase });
+          }}
+          buttonContent="UNDERSTOOD"
+        >
+          Your default database configuration has been changed or updated. Make note of the query you have built before
+          clicking UNDERSTOOD. Clicking UNDERSTOOD will update your query parameters with the new database data.
```

Comment:
Maybe some inline formatting for the button message, like GitHub does for `code` would make more sense here. Also seems like the caps stand out a bit too much. 

I'd probably lean towards being more descriptive with the button action. Something like `Update Query` or similar.